Sources/NIONFS3/NFSTypes+Containers.swift

```diff
@@ -0,0 +1,529 @@
+              let programVersion = self.readInteger(endianness: .big, as: UInt32.self),
+              let procedure = self.readInteger(endianness: .big, as: UInt32.self) else {
+                  throw NFS3Error.illegalRPCTooShort
+              }
+        let credentials = try self.readRPCCredentials()
+        let verifier = try self.readRPCVerifier()
+
+        guard version == 2 else {
+            throw RPCErrors.unknownVersion(version)
+        }
+
+        return RPCCall(xid: xid,
+                       rpcVersion: version,
+                       program: program,
+                       programVersion: programVersion,
+                       procedure: procedure,
+                       credentials: credentials,
+                       verifier: verifier)
+    }
+
+    public mutating func readNFSReply(programAndProcedure: RPCNFSProcedureID, rpcReply: RPCReply) throws -> RPCNFS3Reply {
+        switch programAndProcedure {
+        case .mountNull:
+            return .init(rpcReply: rpcReply, nfsReply: .mountNull)
+        case .mountMount:
+            return .init(rpcReply: rpcReply, nfsReply: .mount(try self.readNFSReplyMount()))
+        case .mountUnmount:
+            return .init(rpcReply: rpcReply, nfsReply: .unmount(try self.readNFSReplyUnmount()))
+        case .nfsNull:
+            return .init(rpcReply: rpcReply, nfsReply: .null)
+        case .nfsGetAttr:
+            return .init(rpcReply: rpcReply, nfsReply: .getattr(try self.readNFSReplyGetAttr()))
+        case .nfsFSInfo:
+            return .init(rpcReply: rpcReply, nfsReply: .fsinfo(try self.readNFSReplyFSInfo()))
+        case .nfsPathConf:
+            return .init(rpcReply: rpcReply, nfsReply: .pathconf(try self.readNFSReplyPathConf()))
+        case .nfsFSStat:
+            return .init(rpcReply: rpcReply, nfsReply: .fsstat(try self.readNFSReplyFSStat()))
+        case .nfsAccess:
+            return .init(rpcReply: rpcReply, nfsReply: .access(try self.readNFSReplyAccess()))
+        case .nfsLookup:
+            return .init(rpcReply: rpcReply, nfsReply: .lookup(try self.readNFSReplyLookup()))
+        case .nfsReadDirPlus:
+            return .init(rpcReply: rpcReply, nfsReply: .readdirplus(try self.readNFSReplyReadDirPlus()))
+        case .nfsReadDir:
+            return .init(rpcReply: rpcReply, nfsReply: .readdir(try self.readNFSReplyReadDir()))
+        case .nfsRead:
+            return .init(rpcReply: rpcReply, nfsReply: .read(try self.readNFSReplyRead()))
+        case .nfsReadLink:
+            return .init(rpcReply: rpcReply, nfsReply: .readlink(try self.readNFSReplyReadlink()))
+        case .nfsSetAttr:
+            return .init(rpcReply: rpcReply, nfsReply: .setattr(try self.readNFSReplySetattr()))
+        default:
+            throw NFS3Error.unknownProgramOrProcedure(.reply(rpcReply))
+        }
+    }
+
+    mutating func readNFSCall(rpc: RPCCall) throws -> RPCNFS3Call {
```

Comment:
Should this return `nil` if there are insufficient bytes?

Reply:
SUN RPC